src/main/java/org/wildfly/security/util/ProviderServiceLoaderSupplier.java

```diff
@@ -39,14 +39,23 @@
  */
 public class ProviderServiceLoaderSupplier extends ServiceLoaderSupplier<Provider> {
 
+    final boolean elytronProviderStaticallyAdded;
```

Comment:
Why was that necessary to introduce?

Reply:
For cases when will be Elytron provider added AFTER - like https://github.com/wildfly-security/wildfly-elytron/pull/1101/files#diff-c326ff349cb9d8a1a54a2f94e9e3683aL167

Reply:
@mchoma This was added because ```ProviderServiceLoaderSupplier``` is now also being used from ```AuthenticationConfiguration#useProvidersFromClassLoader``` and in the handling of ```<use-service-loader module-name=...>```. In these cases, we shouldn't just filter out ```WildFlyElytronProvider``` if it is found since it wasn't already added explicitly by us like it is for the ```DEFAULT_PROVIDER_SUPPLIER```.